source/TagRestrictions/TagsChecker.cs

```diff
@@ -78,25 +80,31 @@ private bool RequiredCheck(HashSet<string> sourceTags, HashSet<string> targetTag
 
         private bool RequiredAnyCheck(HashSet<string> sourceTags, HashSet<string> targetTags)
         {
-            foreach (var tag in sourceTags)
+            if (RequiredAnyCheck(sourceTags, targetTags, RequiredAnyTags)) return true;
```

Comment:
Would rename the existing "shared" `RequiredCheck` to `SharedRequiredCheck` or so... (to distinguish it from the actual `RequiredCheck` method that doesn't have the location parameter ) and do the same for the newly introduced shared `RequiredAnyCheck` method here.